moveit_ros/visualization/planning_scene_rviz_plugin/src/planning_scene_display.cpp

```diff
@@ -491,12 +489,11 @@ void PlanningSceneDisplay::unsetLinkColor(rviz_default_plugins::robot::Robot* ro
 // ******************************************************************************************
 // Load
 // ******************************************************************************************
-planning_scene_monitor::PlanningSceneMonitorPtr
-PlanningSceneDisplay::createPlanningSceneMonitor(const rclcpp::Node::SharedPtr& node)
+planning_scene_monitor::PlanningSceneMonitorPtr PlanningSceneDisplay::createPlanningSceneMonitor()
 {
   std::shared_ptr<tf2_ros::Buffer> tf_buffer = moveit::planning_interface::getSharedTF();
   return std::make_shared<planning_scene_monitor::PlanningSceneMonitor>(
-      node, robot_description_property_->getStdString(), tf_buffer, getNameStd() + "_planning_scene_monitor");
+      node_, robot_description_property_->getStdString(), tf_buffer, getNameStd() + "_planning_scene_monitor");
```

Comment:
If this is the only reason for the node, maybe we should provide a PlanningSceneMonitor constructor that provides its own node

Reply:
The PSM is useless without a node which means that it should probably have one as a member variable. Considering that it's use case is very similar to that of the TF Listener where it should asynchronously subscribe and publish to topics, I think that we should model it after the TF Listener which has a constructor that optionally takes in a node. If you pass in a node, it subscribes and publishes with the node passed in but if you don't provide one, it creates it's own node that spins it with it's own separate executor. See TF Listener + TF Buffer for the full implementation.